pkg/metapage/multi.go

```diff
@@ -163,9 +163,17 @@ func (m *MultiPager) Exists(offset uint64) (bool, error) {
 	return true, nil
 }
 
-func (m *MultiPager) Reset(offset uint64) error {
-	// write a full page of zeros
-	emptyPage := make([]byte, m.rws.PageSize())
+func (m *MultiPager) SlotExists(offset uint64) (bool, error) {
+	return m.spaceExists(offset, m.rws.SlotSize())
+}
+
+func (m *MultiPager) PageExists(offset uint64) (bool, error) {
+	return m.spaceExists(offset, m.rws.PageSize())
+}
+
+func (m *MultiPager) spaceReset(offset uint64, size int) error {
+	// write a full slot of zeros
+	emptyPage := make([]byte, size)
 	binary.LittleEndian.PutUint64(emptyPage[12:20], ^uint64(0))
```

Comment:
Do slots actually need a pointer to the next slot? Or could they just do `offset + 256`?

Reply:
Ah, we don't need a pointer to the next slot. The typescript version ingests a page, splits it into 16 slots, then uses the last slot's next pointer to fetch the next batch of slots.

I guess a possible refactor would be to remove all next pointers, and store them at the page level? I figured for consistency's sake, every slot contains the next pointer info

Reply:
Yeah, I think it can be stored at the page level. Although it doesn't really affect the storage capacity all things considered.